netmiko/tplink/tplink_jetstream.py

```diff
@@ -0,0 +1,153 @@
+        """
+        delay_factor = self.select_delay_factor(delay_factor=0)
+        time.sleep(0.3 * delay_factor)
+        self.clear_buffer()
+        self._test_channel_read(pattern=r"[>#]")
+        self.set_base_prompt()
+        self.enable()
+        self.disable_paging()
+        # Clear the read buffer
+        time.sleep(0.3 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def enable(self, cmd="enable", pattern="ssword", re_flags=re.IGNORECASE):
+        """
+        Enter enable mode.
+
+        If the user does not have the Admin role it will need to execute
+        enable-admin to really enable all functions.
+        """
+        fallback_cmd = "enable-admin"
+        try:
+            return super().enable(cmd=cmd, pattern=pattern, re_flags=re_flags)
+        except ValueError:
+            return super().enable(cmd=fallback_cmd, pattern=pattern, re_flags=re_flags)
+
+    def config_mode(self, config_command="configure"):
+        """Enter configuration mode."""
+        return super().config_mode(config_command=config_command)
+
+    def exit_config_mode(self, exit_config="exit", pattern=r"#"):
+        """
+        Exit config mode.
+
+        Like the Mellanox equipment, the TP-Link Jetstream does not
+        support a single command to completely exit the configuration mode.
+
+        Consequently, need to keep checking and sending "exit".
+        """
+        output = ""
+        check_count = 12
+        while check_count >= 0:
+            if self.check_config_mode():
+                self.write_channel(self.normalize_cmd(exit_config))
+                output += self.read_until_pattern(pattern=pattern)
+            else:
+                break
+            check_count -= 1
+
+        if self.check_config_mode():
+            raise ValueError("Failed to exit configuration mode")
+            log.debug(f"exit_config_mode: {output}")
+
+        return output
+
+    def check_config_mode(self, check_string="(config", pattern=r"#"):
+        """Check whether device is in configuration mode. Return a boolean."""
+        return super().check_config_mode(check_string=check_string, pattern=pattern)
+
+    def set_base_prompt(
+        self, pri_prompt_terminator=">", alt_prompt_terminator="#", delay_factor=1
+    ):
+        """
+        Sets self.base_prompt
+
+        Used as delimiter for stripping of trailing prompt in output.
+
+        Should be set to something that is general and applies in multiple
+        contexts. For TP-Link this will be the router prompt with > or #
+        stripped off.
+
+        This will be set on logging in, but not when entering system-view
+        """
+        prompt = super().set_base_prompt(
+            pri_prompt_terminator=pri_prompt_terminator,
+            alt_prompt_terminator=alt_prompt_terminator,
+            delay_factor=delay_factor,
+        )
+
+        prompt = prompt.strip()
+        self.base_prompt = prompt
+        return self.base_prompt
```

Comment:
This looked a bit odd to me since the parent class already sets `self.base_prompt()` is the extra strip and then resetting `self.base_prompt` necessary?

@piterpunk

Reply:
The lines 102 and 103 can be removed, they are leftovers of the driver that I used as base.